packages/react/src/primitives/thread/useThreadViewportAutoScroll.tsx

```diff
@@ -36,37 +40,173 @@ export const useThreadViewportAutoScroll = <TElement extends HTMLElement>({
     div.scrollTo({ top: div.scrollHeight, behavior });
   }, []);
 
-  const handleScroll = () => {
-    const div = divRef.current;
-    if (!div) return;
+  const updateSpacerForAnchor = useCallback(
+    (element: HTMLElement) => {
+      const div = divRef.current;
+      if (!div) return;
+
+      const spacer = div.querySelector<HTMLElement>(
+        ".aui-thread-viewport-spacer",
```

Comment:
Using a hardcoded CSS class name in our Radix UI components is an antipattern / the class name is hard to discover from the JS API surface. I would rather we find a more elegant way to do this. Let's go through this tomorrow to cleanly separate the API between the library and shadcn templates